### PR TITLE
Preserve sources with no data recorded in virtual overview files

### LINKED ISSUE #285: Virtual overview file missing instrument sources with no data
E.g. `MID_EXP_AGIPD1M1/REDU/LITFRM:output` is

- present in `/gpfs/exfel/exp/MID/202230/p900249/raw/r0198/RAW-R0198-DA05-S00000.h5`
- missing from `/gpfs/exfel/exp/MID/202230/p900249/usr/.extra_data/RAW-R0198-OVERVIEW.h5`

We haven't figured out why yet.

### PR DESCRIPTION
**Pull request by takluyver**
I'm not entirely sure if it's possible to make a virtual dataset with no source data - h5py will just make it a regular dataset instead, so I've opted to make that explicit.

Closes #285.

### CODE REVIEW

File: extra_data/writer.py [hunk 1]
```diff
@@ -257,14 +249,16 @@ def copy_dataset(self, source, key):
         self._make_index(source, key, a.coords['trainId'].values)
 
     def add_dataset(self, source, key):
-        layout, train_ids = self._assemble_data(source, key)
-        if not layout:
-            return  # No data
+        keydata = self.data[source, key]
+        path = f"{keydata.section}/{source}/{key.replace('.', '/')}"
```
Comment (tmichela) [inline extra_data/writer.py:253]:
```suggestion
```

Comment (tmichela) [inline extra_data/writer.py:253]:
```suggestion
```

`KeyData` has a `hdf5_data_path` property that does that.

File: extra_data/writer.py [hunk 2]
```diff
@@ -257,14 +249,16 @@ def copy_dataset(self, source, key):
...
 
-        path = f"{self._section(source)}/{source}/{key.replace('.', '/')}"
-        self.file.create_virtual_dataset(path, layout)
+        if keydata.shape[0] == 0:  # No data
+            self.file.create_dataset(path, shape=keydata.shape, dtype=keydata.dtype)
+        else:
+            layout = self._assemble_data(keydata)
+            self.file.create_virtual_dataset(path, layout)
```
Comment (tmichela) [inline extra_data/writer.py:259]:
```suggestion
        if keydata.shape[0] == 0:  # No data
            self.file.create_dataset(keydata.hdf5_data_path, shape=keydata.shape, dtype=keydata.dtype)
        else:
            layout = self._assemble_data(keydata)
            self.file.create_virtual_dataset(keydata.hdf5_data_path, layout)
```

**Review by tmichela — APPROVED** [review]
LGTM

**Comment by takluyver** [comment]
Thanks, good point about `.hdf5_data_path`.

**Pull request merged**